crates/providers/src/new.rs

```diff
@@ -84,48 +98,63 @@ pub trait Provider<N: Network, T: Transport + Clone = BoxTransport>: Send + Sync
 
     async fn new_pending_transaction(&self, tx_hash: B256) -> TransportResult<PendingTransaction>;
 
-    async fn estimate_gas(&self, tx: &N::TransactionRequest) -> TransportResult<U256> {
-        self.client().prepare("eth_estimateGas", (tx,)).await
-    }
-
     /// Get the last block number available.
     async fn get_block_number(&self) -> TransportResult<BlockNumber> {
         self.client().prepare("eth_blockNumber", ()).await.map(|num: U64| num.to::<u64>())
     }
 
-    /// Get the transaction count for an address. Used for finding the
-    /// appropriate nonce.
-    ///
-    /// TODO: block number/hash/tag
-    async fn get_transaction_count(&self, address: Address) -> TransportResult<U256> {
-        self.client().prepare("eth_getTransactionCount", (address, "latest")).await
+    /// Gets the transaction count of the corresponding address.
+    async fn get_transaction_count(
+        &self,
+        address: Address,
+        tag: Option<BlockId>,
+    ) -> TransportResult<alloy_primitives::U256> {
+        self.client().prepare("eth_getTransactionCount", (address, tag.unwrap_or_default())).await
```

Comment:
personally dont like `prepare` and would like to call it `request`, can do in follow-up

Reply:
https://github.com/alloy-rs/alloy/issues/270